### ISSUE TITLE
Datasets used for training the feature normalizer

### ISSUE THREAD

**Issue by praeclarumjj3** [facebookresearch/large_concept_model#10]
Hi,

Thanks for the great work! I read the paper but could not find information about the dataset mixture used to train the feature normalizer (used inside the PreNet and PostNet).

This repo mentions using a [mixture of datasets](https://github.com/facebookresearch/large_concept_model?tab=readme-ov-file#fitting-a-normalizer) in the readme, but it would be helpful to know the original datasets used in the paper.

I am unsure if a pre-trained normalizer is used when training an LCM or if the entire LCM model is trained end-to-end from scratch, so clarification regarding the above command would be appreciated!

Thanks for the help in advance!

**Comment by elbayadm** [comment]
Hi, @praeclarumjj3!
The sonar normalizer is fitted offline prior to the training of the LCM.
Once part of the LCM's postnet and prenet, its parameters are loaded as frozen buffers
https://github.com/facebookresearch/large_concept_model/blob/main/lcm/models/sonar_normalizer/builder.py#L107-L112

We fit the normalizer using the script provided [here](https://github.com/facebookresearch/large_concept_model/blob/main/scripts/fit_embedding_normalizer.py) which simply estimates a center and a scale from a sample dataset. 
We sample 1M sentences (sonar embeddings) from different domains namely (1) common crawl data (e.g., C4) (2) Books (e.g. Gutenberg) and (3) Wikipedia. 
The goal is to cover a wide range of domains to ensure that the estimated center and scale are not biased to a single domain.
